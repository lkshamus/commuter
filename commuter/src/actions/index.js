export const setCurrentWeather = (weather) => {
  return {
    type: 'SET_CURRENT_WEATHER',
    weather
  }
}

export const isLoading = (bool) => {
  return {
    type: 'IS_LOADING',
    isLoading: bool
  }
}

export const updateSearchWeather = (searchWeather) => {
  return {
    type: 'UPDATE_WEATHER_SEARCH',
    searchWeather
  }
}

export const setDirections = (directions) => {
  return {
    type: 'SET_DIRECTION',
    directions
  }
}

export const updateDirections = (searchedDirections) => {
  return {
    type: 'UPDATE_DIRECTIONS',
  searchedDirections
  }
}

export const setTransitDirections = (transitDirections) => {
  return {
    type: 'SET_TRANSIT_DIRECTIONS', 
    transitDirections
  }
}

export const setBikeDirections = (bikeDirections) => {
  return {
    type: 'SET_BIKE_DIRECTIONS',
    bikeDirections
  }
}

export const setWalkingDirections = (walkingDirections) => {
  return {
    type: 'SET_WALKING_DIRECTIONS', 
    walkingDirections
  }
}