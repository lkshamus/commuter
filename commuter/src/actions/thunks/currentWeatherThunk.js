import fetchCall from '../../utilities/fetchCall';
import * as cleaner from '../../utilities/cleaner';
import { setCurrentWeather } from '../index'

export const getCurrentWeather = () => {
  return async (dispatch) => {
    // dispatch(isLoading(true));
    try {
      const weather = await fetchCall(cleaner.searchWeatherByCity);
      console.log(weather)
      // if (weather === 'failed') {
      //   dispatch(setIsOk(true));
      // }
      // dispatch(isLoading(false));
      return dispatch(setCurrentWeather(weather));
    } catch(error) {
      console.log(error.message)
      // dispatch(setHasErrored(true));
    }
  }
}