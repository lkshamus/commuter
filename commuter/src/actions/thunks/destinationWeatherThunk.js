import fetchCall from '../../utilities/fetchCall';
import * as cleaner from '../../utilities/cleaner';
import { isLoading, updateSearchWeather, setHasErrored, setHasFailed } from '../index'

export const getDestinationWeather = (city) => {
  return async (dispatch) => {
    dispatch(isLoading(true));
    try {
      const weather = await fetchCall(cleaner.hourlyWeatherByCity(city));
      if (weather === 'failed') {
        dispatch(setHasFailed(true));
      }
      dispatch(isLoading(false));
      return dispatch(updateSearchWeather(weather));
    } catch(error) {
      console.log(error.message)
      dispatch(setHasErrored(true));
    }
  }
}
