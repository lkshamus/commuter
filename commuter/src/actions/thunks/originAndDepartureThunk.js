// import fetchCall from '../../utilities/fetchCall';
// import * as cleaner from '../../utilities/cleaner';
// import { isLoading, setDirections } from '../index'

// export const getCurrentDrivingDirections = (origin, departure) => {
//   return async (dispatch) => {
//     dispatch(isLoading(true));
//     try {
//       const directions = await fetchCall(cleaner.orignAndDepartureDriving(origin, departure));
//       console.log('Driving', directions)
//       // if (weather === 'failed') {
//       //   dispatch(setIsOk(true));
//       // }
//       dispatch(isLoading(false));
//       return dispatch(setDirections(directions));
//     } catch(error) {
//       console.log('this is the error block')
//       console.log(error.message)
//       // dispatch(setHasErrored(true));
//     }
//   }
// }
