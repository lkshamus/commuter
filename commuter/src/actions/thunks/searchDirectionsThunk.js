import fetchCall from '../../utilities/fetchCall';
import * as cleaner from '../../utilities/cleaner';
import { isLoading, setDirections, setHasFailed, setHasErrored } from '../index'

export const getCurrentDirections = (origin, destination, mode) => {
  return async (dispatch) => {
    dispatch(isLoading(true));
   try {
      const directions = await fetchCall(cleaner.orignAndDeparture(origin, destination, mode));
      if (directions === 'failed') {
        dispatch(setHasFailed(true));
      }
      dispatch(isLoading(false));
      return dispatch(setDirections(directions));
   } catch(error) {
      console.log('this is the error block')
      console.log(error.message)
     dispatch(setHasErrored(true));
   }
  }
}