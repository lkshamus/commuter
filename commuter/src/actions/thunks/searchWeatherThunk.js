import fetchCall from '../../utilities/fetchCall';
import * as cleaner from '../../utilities/cleaner';
import { isLoading, setCurrentWeather, updateSearchWeather } from '../index'

export const getCurrentWeather = (city) => {
  return async (dispatch) => {
    dispatch(isLoading(true));
    try {
      const weather = await fetchCall(cleaner.searchWeatherByCity(city));
      console.log('weather results:', weather)
      // if (weather === 'failed') {
      //   dispatch(setIsOk(true));
      // }
      dispatch(isLoading(false));
      return dispatch(setCurrentWeather(weather));
    } catch(error) {
      console.log(error.message)
      // dispatch(setHasErrored(true));
    }
  }
}
