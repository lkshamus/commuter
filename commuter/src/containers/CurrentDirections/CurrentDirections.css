.display-directions {
  width: 100vw;
  align-items: center;
  height: 800px;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 400px 1fr;
}
.direction-text {
  /*border: solid 1px;*/
  width: 300px;
  margin-top: 40px;
  display: grid;
  justify-content: center;
  align-items: center;
  /*border-radius: 10%;*/
}

.directions-box {
  display: flex;
  grid-row: 1 / span 1;
  justify-content: center;
  align-items: center;
}

.map {
  text-align: center;
  /*grid-column: 2 / span 1;*/
  /*display: flex;*/
  /*align-items: center;*/
  /*justify-content: center;*/
  margin-top: 50px;
  border-radius: 10%;
}

h3 {
  color: #193446;
  margin: 10px;
  font: 300 18px/1.7 'Open Sans', sans-serif;
}

.text {
  color: #9aabb9;
  font: 600 20px/1.7 'Open Sans', sans-serif;
}

.directions, .directions-weather {
  font-size: 20px;
}