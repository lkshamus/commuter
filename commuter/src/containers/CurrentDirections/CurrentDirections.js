import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import CurrentWeather from '../CurrentWeather/CurrentWeather'
import Loading from '../../components/Loading/index'
import { gmapApi } from '../../utilities/APIkey'
import './CurrentDirections.css'

export class CurrentDirections extends Component {

 render () {
  const startLat = this.props.directions.routes[0].legs[0].start_location.lat
  const startLng = this.props.directions.routes[0].legs[0].start_location.lng
  const endLat = this.props.directions.routes[0].legs[0].end_location.lat
  const endLng = this.props.directions.routes[0].legs[0].end_location.lng
    try{
        return (
          <div>
            <div className='display-directions'>
              <h2>Travel</h2>
              <div className='starting-address'>
                <h3 className='directions'> <span className='text'> Travel Time:</span> <span className='result'>{this.props.directions.routes[0].legs[0].duration.text}</span></h3>  
                <h3 className='directions'> <span className='text'>Starting Address: </span> <span className='result'>{this.props.directions.routes[0].legs[0].start_address}</span></h3>                      
              </div>
              <div className='ending-address'>
                <h3 className='directions'> <span className='text'> Distance in Miles:</span> <span className='result'>{this.props.directions.routes[0].legs[0].distance.text}</span></h3>            
                <h3 className='directions'> <span className='text'> Ending Address:</span><span className='result'> {this.props.directions.routes[0].legs[0].end_address}</span></h3>
                </div>
              <CurrentWeather />
            <div className='map'>
              <img alt='map' className ='mapping' src={`https://maps.googleapis.com/maps/api/staticmap?size=512x512&maptype=roadmap\ &markers=size:mid%7Ccolor:green%7C${startLat}, ${startLng}%7C&markers=size:mid%7Ccolor:red%7C${endLat}, ${endLng}&key=${gmapApi}`}/>
            </div>            
          </div>
          </div>
        )
    } catch {
        return (
          <Loading />
        )
    }
  }
}

export const mapStateToProps = (state) => ({
  directions: state.directions,
  isLoading: state.isLoading
});

export default connect(mapStateToProps, null)(CurrentDirections);

CurrentDirections.propTypes = {
  directions: PropTypes.object, 
  isLoading: PropTypes.bool,
};