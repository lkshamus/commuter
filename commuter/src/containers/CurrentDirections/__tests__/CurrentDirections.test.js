/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { connect } from 'react-redux';

import { getCurrentDirections } from '../../../actions/thunks/searchDirectionsThunk'
import { CurrentDirections, mapDispatchToProps, mapStateToProps } from '../CurrentDirections';
import { setDirections } from '../../../actions'
import { setCurrentWeather } from '../../../actions'
import { updateSearchWeather } from '../../../actions'

describe('CurrentDirections', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<CurrentDirections />)
    expect(wrapper).toMatchSnapshot();
  })
})

describe('mapStateToProps', () => {
  it('should map the state to props', () => {

    const directions = {origin: 'Mountain View', destination: 'San Jose'}

    const expected = {directions}

    const mockState = {
      directions: directions
    }
    
    const mappedProps = mapStateToProps(mockState)
    expect (mappedProps).toEqual(expected)

  })
})
