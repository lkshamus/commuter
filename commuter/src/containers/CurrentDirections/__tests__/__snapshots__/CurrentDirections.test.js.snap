// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentDirections should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrentDirections
    directions={
      Object {
        "geocoded_waypoints": Array [
          Object {
            "geocoder_status": "OK",
            "place_id": "ChIJa147K9HX3IAR-lwiGIQv9i4",
            "types": Array [
              "amusement_park",
              "establishment",
              "point_of_interest",
            ],
          },
          Object {
            "geocoder_status": "OK",
            "place_id": "ChIJzzgyJU--woARcZqceSdQ3dM",
            "types": Array [
              "amusement_park",
              "establishment",
              "point_of_interest",
            ],
          },
        ],
        "routes": Array [
          Object {
            "bounds": Object {
              "northeast": Object {
                "lat": 34.1373841,
                "lng": -117.9220826,
              },
              "southwest": Object {
                "lat": 33.8151707,
                "lng": -118.3575585,
              },
            },
            "copyrights": "Map data Â©2018 Google",
            "legs": Array [
              Object {
                "distance": Object {
                  "text": "36.0 mi",
                  "value": 57975,
                },
                "duration": Object {
                  "text": "51 mins",
                  "value": 3078,
                },
                "end_address": "100 Universal City Plaza, Universal City, CA 91608, USA",
                "end_location": Object {
                  "lat": 34.1364887,
                  "lng": -118.3569926,
                },
                "start_address": "1313 Disneyland Dr, Anaheim, CA 92802, USA",
                "start_location": Object {
                  "lat": 33.8162219,
                  "lng": -117.9224731,
                },
                "steps": Array [
                  Object {
                    "distance": Object {
                      "text": "59 ft",
                      "value": 18,
                    },
                    "duration": Object {
                      "text": "1 min",
                      "value": 2,
                    },
                    "end_location": Object {
                      "lat": 33.8160679,
                      "lng": -117.9225314,
                    },
                    "html_instructions": "Head <b>south</b>",
                    "polyline": Object {
                      "points": "kvkmElvvnUJ",
                    },
                    "start_location": Object {
                      "lat": 33.8162219,
                      "lng": -117.9224731,
                    },
                    "travel_mode": "DRIVING",
                  },
                ],
              },
            ],
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="display-directions"
      >
        <h2>
          Travel
        </h2>
        <div
          className="starting-address"
        >
          <h3
            className="directions"
          >
             
            <span
              className="text"
            >
               Travel Time:
            </span>
             
            <span
              className="result"
            >
              51 mins
            </span>
          </h3>
          <h3
            className="directions"
          >
             
            <span
              className="text"
            >
              Starting Address: 
            </span>
             
            <span
              className="result"
            >
              1313 Disneyland Dr, Anaheim, CA 92802, USA
            </span>
          </h3>
        </div>
        <div
          className="ending-address"
        >
          <h3
            className="directions"
          >
             
            <span
              className="text"
            >
               Distance in Miles:
            </span>
             
            <span
              className="result"
            >
              36.0 mi
            </span>
          </h3>
          <h3
            className="directions"
          >
             
            <span
              className="text"
            >
               Ending Address:
            </span>
            <span
              className="result"
            >
               
              100 Universal City Plaza, Universal City, CA 91608, USA
            </span>
          </h3>
        </div>
        <Connect(CurrentWeather) />
        <div
          className="map"
        >
          <img
            alt="map"
            className="mapping"
            src="https://maps.googleapis.com/maps/api/staticmap?size=512x512&maptype=roadmap &markers=size:mid%7Ccolor:green%7C33.8162219, -117.9224731%7C&markers=size:mid%7Ccolor:red%7C34.1364887, -118.3569926&key=AIzaSyA82too8VL24aPrNvt2kJhfLOpM-pGmXrY"
          />
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Travel
          </h2>,
          <div
            className="starting-address"
          >
            <h3
              className="directions"
            >
               
              <span
                className="text"
              >
                 Travel Time:
              </span>
               
              <span
                className="result"
              >
                51 mins
              </span>
            </h3>
            <h3
              className="directions"
            >
               
              <span
                className="text"
              >
                Starting Address: 
              </span>
               
              <span
                className="result"
              >
                1313 Disneyland Dr, Anaheim, CA 92802, USA
              </span>
            </h3>
          </div>,
          <div
            className="ending-address"
          >
            <h3
              className="directions"
            >
               
              <span
                className="text"
              >
                 Distance in Miles:
              </span>
               
              <span
                className="result"
              >
                36.0 mi
              </span>
            </h3>
            <h3
              className="directions"
            >
               
              <span
                className="text"
              >
                 Ending Address:
              </span>
              <span
                className="result"
              >
                 
                100 Universal City Plaza, Universal City, CA 91608, USA
              </span>
            </h3>
          </div>,
          <Connect(CurrentWeather) />,
          <div
            className="map"
          >
            <img
              alt="map"
              className="mapping"
              src="https://maps.googleapis.com/maps/api/staticmap?size=512x512&maptype=roadmap &markers=size:mid%7Ccolor:green%7C33.8162219, -117.9224731%7C&markers=size:mid%7Ccolor:red%7C34.1364887, -118.3569926&key=AIzaSyA82too8VL24aPrNvt2kJhfLOpM-pGmXrY"
            />
          </div>,
        ],
        "className": "display-directions",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Travel",
          },
          "ref": null,
          "rendered": "Travel",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="directions"
              >
                 
                <span
                  className="text"
                >
                   Travel Time:
                </span>
                 
                <span
                  className="result"
                >
                  51 mins
                </span>
              </h3>,
              <h3
                className="directions"
              >
                 
                <span
                  className="text"
                >
                  Starting Address: 
                </span>
                 
                <span
                  className="result"
                >
                  1313 Disneyland Dr, Anaheim, CA 92802, USA
                </span>
              </h3>,
            ],
            "className": "starting-address",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="text"
                  >
                     Travel Time:
                  </span>,
                  " ",
                  <span
                    className="result"
                  >
                    51 mins
                  </span>,
                ],
                "className": "directions",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Travel Time:",
                    "className": "text",
                  },
                  "ref": null,
                  "rendered": " Travel Time:",
                  "type": "span",
                },
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "51 mins",
                    "className": "result",
                  },
                  "ref": null,
                  "rendered": "51 mins",
                  "type": "span",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="text"
                  >
                    Starting Address: 
                  </span>,
                  " ",
                  <span
                    className="result"
                  >
                    1313 Disneyland Dr, Anaheim, CA 92802, USA
                  </span>,
                ],
                "className": "directions",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Starting Address: ",
                    "className": "text",
                  },
                  "ref": null,
                  "rendered": "Starting Address: ",
                  "type": "span",
                },
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1313 Disneyland Dr, Anaheim, CA 92802, USA",
                    "className": "result",
                  },
                  "ref": null,
                  "rendered": "1313 Disneyland Dr, Anaheim, CA 92802, USA",
                  "type": "span",
                },
              ],
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="directions"
              >
                 
                <span
                  className="text"
                >
                   Distance in Miles:
                </span>
                 
                <span
                  className="result"
                >
                  36.0 mi
                </span>
              </h3>,
              <h3
                className="directions"
              >
                 
                <span
                  className="text"
                >
                   Ending Address:
                </span>
                <span
                  className="result"
                >
                   
                  100 Universal City Plaza, Universal City, CA 91608, USA
                </span>
              </h3>,
            ],
            "className": "ending-address",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="text"
                  >
                     Distance in Miles:
                  </span>,
                  " ",
                  <span
                    className="result"
                  >
                    36.0 mi
                  </span>,
                ],
                "className": "directions",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Distance in Miles:",
                    "className": "text",
                  },
                  "ref": null,
                  "rendered": " Distance in Miles:",
                  "type": "span",
                },
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "36.0 mi",
                    "className": "result",
                  },
                  "ref": null,
                  "rendered": "36.0 mi",
                  "type": "span",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="text"
                  >
                     Ending Address:
                  </span>,
                  <span
                    className="result"
                  >
                     
                    100 Universal City Plaza, Universal City, CA 91608, USA
                  </span>,
                ],
                "className": "directions",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Ending Address:",
                    "className": "text",
                  },
                  "ref": null,
                  "rendered": " Ending Address:",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "100 Universal City Plaza, Universal City, CA 91608, USA",
                    ],
                    "className": "result",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "100 Universal City Plaza, Universal City, CA 91608, USA",
                  ],
                  "type": "span",
                },
              ],
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="map"
              className="mapping"
              src="https://maps.googleapis.com/maps/api/staticmap?size=512x512&maptype=roadmap &markers=size:mid%7Ccolor:green%7C33.8162219, -117.9224731%7C&markers=size:mid%7Ccolor:red%7C34.1364887, -118.3569926&key=AIzaSyA82too8VL24aPrNvt2kJhfLOpM-pGmXrY"
            />,
            "className": "map",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "map",
              "className": "mapping",
              "src": "https://maps.googleapis.com/maps/api/staticmap?size=512x512&maptype=roadmap &markers=size:mid%7Ccolor:green%7C33.8162219, -117.9224731%7C&markers=size:mid%7Ccolor:red%7C34.1364887, -118.3569926&key=AIzaSyA82too8VL24aPrNvt2kJhfLOpM-pGmXrY",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="display-directions"
        >
          <h2>
            Travel
          </h2>
          <div
            className="starting-address"
          >
            <h3
              className="directions"
            >
               
              <span
                className="text"
              >
                 Travel Time:
              </span>
               
              <span
                className="result"
              >
                51 mins
              </span>
            </h3>
            <h3
              className="directions"
            >
               
              <span
                className="text"
              >
                Starting Address: 
              </span>
               
              <span
                className="result"
              >
                1313 Disneyland Dr, Anaheim, CA 92802, USA
              </span>
            </h3>
          </div>
          <div
            className="ending-address"
          >
            <h3
              className="directions"
            >
               
              <span
                className="text"
              >
                 Distance in Miles:
              </span>
               
              <span
                className="result"
              >
                36.0 mi
              </span>
            </h3>
            <h3
              className="directions"
            >
               
              <span
                className="text"
              >
                 Ending Address:
              </span>
              <span
                className="result"
              >
                 
                100 Universal City Plaza, Universal City, CA 91608, USA
              </span>
            </h3>
          </div>
          <Connect(CurrentWeather) />
          <div
            className="map"
          >
            <img
              alt="map"
              className="mapping"
              src="https://maps.googleapis.com/maps/api/staticmap?size=512x512&maptype=roadmap &markers=size:mid%7Ccolor:green%7C33.8162219, -117.9224731%7C&markers=size:mid%7Ccolor:red%7C34.1364887, -118.3569926&key=AIzaSyA82too8VL24aPrNvt2kJhfLOpM-pGmXrY"
            />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Travel
            </h2>,
            <div
              className="starting-address"
            >
              <h3
                className="directions"
              >
                 
                <span
                  className="text"
                >
                   Travel Time:
                </span>
                 
                <span
                  className="result"
                >
                  51 mins
                </span>
              </h3>
              <h3
                className="directions"
              >
                 
                <span
                  className="text"
                >
                  Starting Address: 
                </span>
                 
                <span
                  className="result"
                >
                  1313 Disneyland Dr, Anaheim, CA 92802, USA
                </span>
              </h3>
            </div>,
            <div
              className="ending-address"
            >
              <h3
                className="directions"
              >
                 
                <span
                  className="text"
                >
                   Distance in Miles:
                </span>
                 
                <span
                  className="result"
                >
                  36.0 mi
                </span>
              </h3>
              <h3
                className="directions"
              >
                 
                <span
                  className="text"
                >
                   Ending Address:
                </span>
                <span
                  className="result"
                >
                   
                  100 Universal City Plaza, Universal City, CA 91608, USA
                </span>
              </h3>
            </div>,
            <Connect(CurrentWeather) />,
            <div
              className="map"
            >
              <img
                alt="map"
                className="mapping"
                src="https://maps.googleapis.com/maps/api/staticmap?size=512x512&maptype=roadmap &markers=size:mid%7Ccolor:green%7C33.8162219, -117.9224731%7C&markers=size:mid%7Ccolor:red%7C34.1364887, -118.3569926&key=AIzaSyA82too8VL24aPrNvt2kJhfLOpM-pGmXrY"
              />
            </div>,
          ],
          "className": "display-directions",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Travel",
            },
            "ref": null,
            "rendered": "Travel",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="directions"
                >
                   
                  <span
                    className="text"
                  >
                     Travel Time:
                  </span>
                   
                  <span
                    className="result"
                  >
                    51 mins
                  </span>
                </h3>,
                <h3
                  className="directions"
                >
                   
                  <span
                    className="text"
                  >
                    Starting Address: 
                  </span>
                   
                  <span
                    className="result"
                  >
                    1313 Disneyland Dr, Anaheim, CA 92802, USA
                  </span>
                </h3>,
              ],
              "className": "starting-address",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <span
                      className="text"
                    >
                       Travel Time:
                    </span>,
                    " ",
                    <span
                      className="result"
                    >
                      51 mins
                    </span>,
                  ],
                  "className": "directions",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Travel Time:",
                      "className": "text",
                    },
                    "ref": null,
                    "rendered": " Travel Time:",
                    "type": "span",
                  },
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "51 mins",
                      "className": "result",
                    },
                    "ref": null,
                    "rendered": "51 mins",
                    "type": "span",
                  },
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <span
                      className="text"
                    >
                      Starting Address: 
                    </span>,
                    " ",
                    <span
                      className="result"
                    >
                      1313 Disneyland Dr, Anaheim, CA 92802, USA
                    </span>,
                  ],
                  "className": "directions",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Starting Address: ",
                      "className": "text",
                    },
                    "ref": null,
                    "rendered": "Starting Address: ",
                    "type": "span",
                  },
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "1313 Disneyland Dr, Anaheim, CA 92802, USA",
                      "className": "result",
                    },
                    "ref": null,
                    "rendered": "1313 Disneyland Dr, Anaheim, CA 92802, USA",
                    "type": "span",
                  },
                ],
                "type": "h3",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="directions"
                >
                   
                  <span
                    className="text"
                  >
                     Distance in Miles:
                  </span>
                   
                  <span
                    className="result"
                  >
                    36.0 mi
                  </span>
                </h3>,
                <h3
                  className="directions"
                >
                   
                  <span
                    className="text"
                  >
                     Ending Address:
                  </span>
                  <span
                    className="result"
                  >
                     
                    100 Universal City Plaza, Universal City, CA 91608, USA
                  </span>
                </h3>,
              ],
              "className": "ending-address",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <span
                      className="text"
                    >
                       Distance in Miles:
                    </span>,
                    " ",
                    <span
                      className="result"
                    >
                      36.0 mi
                    </span>,
                  ],
                  "className": "directions",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Distance in Miles:",
                      "className": "text",
                    },
                    "ref": null,
                    "rendered": " Distance in Miles:",
                    "type": "span",
                  },
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "36.0 mi",
                      "className": "result",
                    },
                    "ref": null,
                    "rendered": "36.0 mi",
                    "type": "span",
                  },
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <span
                      className="text"
                    >
                       Ending Address:
                    </span>,
                    <span
                      className="result"
                    >
                       
                      100 Universal City Plaza, Universal City, CA 91608, USA
                    </span>,
                  ],
                  "className": "directions",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Ending Address:",
                      "className": "text",
                    },
                    "ref": null,
                    "rendered": " Ending Address:",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "100 Universal City Plaza, Universal City, CA 91608, USA",
                      ],
                      "className": "result",
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "100 Universal City Plaza, Universal City, CA 91608, USA",
                    ],
                    "type": "span",
                  },
                ],
                "type": "h3",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="map"
                className="mapping"
                src="https://maps.googleapis.com/maps/api/staticmap?size=512x512&maptype=roadmap &markers=size:mid%7Ccolor:green%7C33.8162219, -117.9224731%7C&markers=size:mid%7Ccolor:red%7C34.1364887, -118.3569926&key=AIzaSyA82too8VL24aPrNvt2kJhfLOpM-pGmXrY"
              />,
              "className": "map",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "map",
                "className": "mapping",
                "src": "https://maps.googleapis.com/maps/api/staticmap?size=512x512&maptype=roadmap &markers=size:mid%7Ccolor:green%7C33.8162219, -117.9224731%7C&markers=size:mid%7Ccolor:red%7C34.1364887, -118.3569926&key=AIzaSyA82too8VL24aPrNvt2kJhfLOpM-pGmXrY",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
